src/renderer/src/routes/modals/AddAddress/Step1.tsx

```diff
@@ -0,0 +1,117 @@
+import { useMemo, useState } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import { TbPlus } from 'react-icons/tb'
+import { TBlockchainServiceKey } from '@renderer/@types/blockchain'
+import { TContactAddress } from '@renderer/@types/store'
+import { BlockchainIcon } from '@renderer/components/BlockchainIcon'
+import { Button } from '@renderer/components/Button'
+import { Input } from '@renderer/components/Input'
+import { Separator } from '@renderer/components/Separator'
+import { useBsAggregatorSelector } from '@renderer/hooks/useBlockchainSelector'
+import { useModalLocation, useModalNavigate } from '@renderer/hooks/useModalRouter'
+import { ModalLayout } from '@renderer/layouts/Modal'
+
+type TFormData = {
+  blockchain: TBlockchainServiceKey
+}
+
+type TLocationState = {
+  contactName: string
+  address?: TContactAddress | null
+  handleAddAddress: (address: TContactAddress, addressToEdit?: TContactAddress) => void
+}
+
+export const AddAddressModalStep1 = () => {
+  const { t } = useTranslation('modals', { keyPrefix: 'addAddressStep1' })
+  const { t: commonT } = useTranslation('common', { keyPrefix: 'general' })
+  const { t: blockchainT } = useTranslation('common', { keyPrefix: 'blockchain' })
+  const location = useModalLocation<TLocationState>()
+  const contactName = location.state.contactName
+  const { bsAggregatorRef } = useBsAggregatorSelector()
+  const { modalNavigate } = useModalNavigate()
+  const blockchainServices = Object.keys(bsAggregatorRef.current.blockchainServicesByName) as TBlockchainServiceKey[]
+  const [selectedBlockchain, setSelectedBlockchain] = useState(location.state?.address?.blockchain)
+  const [search, setSearch] = useState<string | null>(null)
+  const addressCopy = location.state.address
```

Comment:
Try to separate the useState, useMemo, hooks and simple const with space, it turns the code easier to read.

```suggestion
  const { t } = useTranslation('modals', { keyPrefix: 'addAddressStep1' })
  const { t: commonT } = useTranslation('common', { keyPrefix: 'general' })
  const { t: blockchainT } = useTranslation('common', { keyPrefix: 'blockchain' })
  const location = useModalLocation<TLocationState>()
  const { bsAggregatorRef } = useBsAggregatorSelector()
  const { modalNavigate } = useModalNavigate()
  
  const [selectedBlockchain, setSelectedBlockchain] = useState(location.state?.address?.blockchain)
  const [search, setSearch] = useState<string | null>(null)
  
    const blockchainServices = Object.keys(bsAggregatorRef.current.blockchainServicesByName) as TBlockchainServiceKey[]
  const addressCopy = location.state.address
    const contactName = location.state.contactName
```

Sometimes you will need a different separation, because some state are using some const, so you will need to declare the const before the state. But always try to separate